File: rust/src/models/config_feature_type.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::config_value_type::ConfigValueType;

/// A static, typed configuration value. No metric — resolved synchronously.
#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct ConfigFeatureType {
    /// Allowed values when `value_type = SELECT`. Empty otherwise.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub options: Option<Vec<String>>,

    /// The feature's value type, fixed at creation.
    pub value_type: ConfigValueType,
}

impl ConfigFeatureType {
    pub fn new(value_type: ConfigValueType) -> Self {
        Self {
            options: None,
            value_type,
        }
    }
}

File: typescript/src/models/configFeatureType.ts
// this file is @generated
import { type ConfigValueType, ConfigValueTypeSerializer } from "./configValueType";
/** A static, typed configuration value. No metric — resolved synchronously. */
export interface ConfigFeatureType {
  /** Allowed values when `value_type = SELECT`. Empty otherwise. */
  options?: string[];

  /** The feature's value type, fixed at creation. */
  valueType: ConfigValueType;
}

export const ConfigFeatureTypeSerializer = {
  _fromJsonObject(object: any): ConfigFeatureType {
    return {
      options: object["options"],
      valueType: ConfigValueTypeSerializer._fromJsonObject(object["value_type"]),
    };
  },

  _toJsonObject(self: ConfigFeatureType): any {
    return {
      options: self.options,
      value_type: ConfigValueTypeSerializer._toJsonObject(self.valueType),
    };
  },
};
